perun-core/src/test/java/cz/metacentrum/perun/core/entry/MembersManagerEntryIntegrationTest.java

```diff
@@ -3037,6 +3037,92 @@ public void getAllMembers_twoMembersFromOneVo() throws Exception {
 			.contains(member1, member2);
 	}
 
+	@Test
+	public void addMemberToMemberVo() throws Exception {
+		System.out.println(CLASS_NAME + "addMemberToMemberVo");
+
+		Vo memberVo = setUpVo("member");
+		Vo parentVo = setUpVo("parent");
+
+		perun.getVosManagerBl().addMemberVo(sess, parentVo, memberVo);
+
+		Member member = setUpMember(memberVo);
+		Member newMember = perun.getMembersManagerBl().getMemberByUser(sess, parentVo, perun.getUsersManagerBl().getUserByMember(sess, member));
+		Attribute attribute = perun.getAttributesManager().getAttribute(sess, newMember, AttributesManager.NS_MEMBER_ATTR_DEF + ":memberOrganizations");
+		ArrayList<String> currentValue = (ArrayList<String>) attribute.getValue();
+
+		assertThat(currentValue).contains(memberVo.getShortName());
+		assertThat(newMember.getUserId()).isEqualTo(member.getUserId());
+	}
+
+	@Test
+	public void removeMemberFromMemberVo() throws Exception {
+		System.out.println(CLASS_NAME + "removeMemberFromMemberVo");
+
+		Vo memberVo1 = setUpVo("member1");
+		Vo memberVo2 = setUpVo("member2");
+		Vo parentVo = setUpVo("parent");
+
+		Member member1 = setUpMember(memberVo1);
+		Member member2 = perun.getMembersManagerBl().createMember(sess, memberVo2, perun.getUsersManagerBl().getUserByMember(sess, member1));
+		perun.getMembersManagerBl().validateMemberAsync(sess, member2);
+
+		perun.getVosManagerBl().addMemberVo(sess, parentVo, memberVo1);
+		perun.getVosManagerBl().addMemberVo(sess, parentVo, memberVo2);
+
+		Member newMember = perun.getMembersManagerBl().getMemberByUser(sess, parentVo, perun.getUsersManagerBl().getUserByMember(sess, member1));
+
+		perun.getMembersManagerBl().deleteMember(sess, member1);
+
+		Attribute attribute = perun.getAttributesManager().getAttribute(sess, newMember, AttributesManager.NS_MEMBER_ATTR_DEF + ":memberOrganizations");
+		ArrayList<String> currentValue = (ArrayList<String>) attribute.getValue();
+
+		assertThat(currentValue).containsOnly(memberVo2.getShortName());
+		assertThat(perun.getMembersManagerBl().getMembers(sess, parentVo).size()).isEqualTo(1);
+
+	}
+
+	@Test
+	public void changeStatusParentVoMembershipChange() throws Exception {
+		System.out.println(CLASS_NAME + "changeStatusParentVoMembershipChange");
+
+		Vo memberVo = setUpVo("member");
+		Vo parentVo = setUpVo("parent");
+
+		Member member = setUpMember(memberVo);
+
+		perun.getVosManagerBl().addMemberVo(sess, parentVo, memberVo);
+
+		assertThat(perun.getMembersManagerBl().getMembers(sess, parentVo).size()).isEqualTo(1);
+
+		perun.getMembersManagerBl().disableMember(sess, member);
+
+		assertThat(perun.getMembersManagerBl().getMembers(sess, parentVo).size()).isEqualTo(0);
```

Comment:
I think this should be 1 because the member in the parentVo shouldn't be completely deleted, but he should start its lifecycle instead. That's why this test fails. You should probably disable the member in the memberVo and then create a relationship between memberVo and parentVo, then the size should be 0, then validate the member, then the size should be 1.

Reply:
Yeah, my bad. Wrote the tests waay before we cleared up how the lifecycle should work and then forgot to change them.